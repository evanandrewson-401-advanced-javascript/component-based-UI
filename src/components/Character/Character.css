.Character {
    margin: 1em;
    position: relative;    
    h1 {
        position: absolute;
        top: 230px;
        height: 70px;
        width: 300px;
        background-color: black;
        opacity: .6;
        margin: 0;
        color: white;
        box-sizing: border-box;
        display: flex;
        justify-content: center;
        align-items: center;
    }
    .species {
        display: flex;
        width: 300px;
        background-color:black;
        opacity: .6;
        justify-content: space-between;
        margin: 0;
        padding: 5px;
        box-sizing: border-box;
        font-size: 1.5rem;

        #left {
            color: white;
        }

        #right {
            color: orange;
        }
    }
    img {
        margin: 0;
    }
}