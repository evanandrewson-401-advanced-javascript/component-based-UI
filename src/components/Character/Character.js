import React from 'react';
import PropTypes from 'prop-types';

const Character = (image, name, species) => (
  <>
    <img src={image} />
    <h1>{name}</h1>
    <p>{species}</p>
  </>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired
}

export default Character;