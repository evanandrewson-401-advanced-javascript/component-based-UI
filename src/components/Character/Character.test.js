import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const character = {
      name: "Rick Sanchez",
      species: "Human",
      image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      };
    const wrapper = shallow(
    <Character
      name = {character.name}
      species = {character.species}
      image = {character.image}
       />);
    expect(wrapper).toMatchSnapshot();
  });
});