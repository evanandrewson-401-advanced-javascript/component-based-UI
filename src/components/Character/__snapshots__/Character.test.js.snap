// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<Fragment>
  <img
    src={
      Object {
        "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        "name": "Rick Sanchez",
        "species": "Human",
      }
    }
  />
  <h1>
    <Component />
  </h1>
  <p />
</Fragment>
`;
